drivers/ethernet/eth_mcux.c

```diff
@@ -555,16 +555,12 @@ static void eth_mcux_phy_event(struct eth_context *context)
 #endif
 			context->link_up = link_up;
 			context->phy_state = eth_mcux_phy_state_read_duplex;
-
-			/* Network interface might be NULL at this point */
-			if (context->iface) {
-				net_eth_carrier_on(context->iface);
-				k_msleep(USEC_PER_MSEC);
-			}
+			net_eth_carrier_on(context->iface);
+			k_msleep(1);
```

Comment:
I'm not sure the entire delay of 1 second is needed here, but I have some reservations about changing this delay from
1 second to 1 ms. The original introduction of this delay was in #12465, so it does seem it solved an issue at the time. Is there a reason you reduced this delay value?

Reply:
Hey @danieldegrasse, the delay you are referring to was already 1 ms. It was in (wrongly) set to 1 second in #24864.

Reply:
Ah ok, I didn't realize `k_sleep` could a timeout in microseconds (maybe it could at that point in Zephyr's history). In that case, glad we fixed this up.